### PR TITLE
SNOW-181550 Fix the installation section and make changes to the instructions for using the driver

### PR DESCRIPTION
**Pull request by sfc-gh-dyoshinaga**
# Overview

SNOW-181550

Description
Fix the installation section and make changes to the instructions on using the driver
- Replace the PHP build instructions with a pointer to the PHP docs, which includes additional information about
  prerequisites (e.g. libxml2).
- Clean up some of the sections under using the driver.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -19,9 +19,13 @@ Prerequisites
...
 In order to build the Snowflake PHP PDO Driver, you must have the following software installed:
 
-- (Windows) Visual Studio
-- (Linux) gcc 5.2 or higher
-- (macOS) clang
+- On Windows: Visual Studio
+- On Linux:
+
+  - gcc 5.2 or higher
+  - cmake
```
Comment (sfc-gh-dyoshinaga) [inline README.rst:26]: On CentOS, I had to install cmake.

```
[parallels@localhost pdo_snowflake]$ ./scripts/build_pdo_snowflake.sh
which: no cmake3 in (/home/parallels/.local/bin:/home/parallels/bin:/home/parallels/.local/bin:/home/parallels/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin)
which: no cmake in (/home/parallels/.local/bin:/home/parallels/bin:/home/parallels/.local/bin:/home/parallels/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin)
```

File: README.rst [hunk 2]
```diff
@@ -105,6 +79,8 @@ Building the Driver on Linux and macOS
 
        $PHP_HOME/bin/php -dextension=modules/pdo_snowflake.so -m | grep pdo_snowflake
```
Comment (sfc-gh-dyoshinaga) [inline README.rst:80]: Are there any additional instructions needed for this on CentOS?

This worked fine on Ubuntu, but on CentOS, the following error occurred:

```
[parallels@localhost pdo_snowflake]$ $PHP_HOME/bin/php -dextension=modules/pdo_snowflake.so -m | grep pdo_snowflake
PHP Warning:  PHP Startup: Unable to load dynamic library 'modules/pdo_snowflake.so' (tried: modules/pdo_snowflake.so (modules/pdo_snowflake.so: undefined symbol: uuid_unparse_lower), /usr/lib64/php/modules/modules/pdo_snowflake.so.so (/usr/lib64/php/modules/modules/pdo_snowflake.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
```

This was for the case where I downloaded and installed PHP.

If I build PHP instead and I use that built version, this error does not occur.

File: README.rst [hunk 3]
```diff
@@ -183,6 +159,12 @@ Installing the Driver on Linux and macOS
...
    that contains the :code:`pdo.so` file).
 
+   To find the PHP extension directory, run:
+
+   .. code-block:: bash
+
+       $PHP_HOME/bin/php -i | grep '^extension_dir'
+
 #. Copy :code:`cacert.pem` from the :code:`libsnowflakeclient` subdirectory in the repository to the directory containing the
    PHP configuration files (e.g. :code:`/etc/php/7.2/fpm/conf.d` for PHP-FPM version 7.2 on Ubuntu).
```
Comment (sfc-gh-dyoshinaga) [inline README.rst:169]: Is it worth mentioning that if users want to test this through the CLI, they should add the configuration files to `/etc/php/7.2/cli/conf.d/`?

File: README.rst [hunk 4]
```diff
@@ -280,6 +274,17 @@ Here is an example of fetch a row:
     }
     $dbh = null;
     echo "OK\n";
+  $>
+
+Limitations in the Driver
```
Comment (sfc-gh-dyoshinaga) [inline README.rst:279]: This section already exists -- I just moved it to the end of the "Using" section.

File: README.rst (hunk 2 above)
Comment (SimbaGithub) [reply README.rst:80]: I was build PHP from source code on Centos so didn't face such issue. I'll get a clean Centos to reproduce. At the same time, could you add following linelibsnowflakeclient/deps-build/linux/uuid/lib/libuuid.a \
after line 66 of scripts/build_pdo_snowflake.sh to see how it works?

Comment (sfc-gh-dyoshinaga) [reply README.rst:80]: Thanks!

I added that line, ran the script again, and ran the command to load the driver.
The command succeeded without any errors.

Comment (SimbaGithub) [reply README.rst:80]: I've added this line in https://github.com/snowflakedb/pdo_snowflake/pull/231

**Review by SimbaGithub — APPROVED** [review]
LGTM

**Pull request merged**